presto-common/src/main/java/com/facebook/presto/common/zorder/ZOrderIntegers.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.zorder;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+public class ZOrderIntegers
+        implements ZOrder
+{
+    public static final int MAX_INPUT_DIMENSIONS = 10;
+
+    // Assume encodingBits is 1-based
+    private List<Integer> encodingBits;
+    private int totalBitLength;
+    private int maxBitLength;
+
+    public ZOrderIntegers(List<Integer> encodingBits)
+    {
+        this.encodingBits = encodingBits;
+
+        totalBitLength = encodingBits.stream().mapToInt(Integer::intValue).sum();
+        maxBitLength = encodingBits.stream().mapToInt(Integer::intValue).max().getAsInt();
+    }
+
+    private void checkInputValidity(List<Integer> input)
+            throws IllegalArgumentException
+    {
+        if (input.size() == 0) {
+            throw new IllegalArgumentException("Input list size should be greater than zero.");
+        }
+
+        if (input.size() > MAX_INPUT_DIMENSIONS) {
+            throw new IllegalArgumentException("Our current Z-Ordering implementation does not support more than 10 input numbers.");
+        }
+
+        if (input.size() > encodingBits.size()) {
+            throw new IllegalArgumentException(format("Input list size (%d) does not match encoding bits list size (%d).",
+                    input.size(), encodingBits.size()));
+        }
+
+        for (int index = 0; index < input.size(); index++) {
+            if (input.get(index) < 0) {
+                throw new IllegalArgumentException("Our current Z-Ordering implementation does not support negative input numbers.");
+            }
+        }
+    }
+
+    private void checkInputValidity(List<Integer> input, Type type)
+            throws IllegalArgumentException
+    {
+        checkInputValidity(input);
+
+        int maximumBits = 0;
+        switch (type) {
+            case INT:
+                maximumBits = 32;
+                break;
+            case LONG:
+                maximumBits = 64;
+                break;
+            default:
+                break;
+        }
+
+        if (totalBitLength > maximumBits) {
+            throw new IllegalArgumentException(format("The z-address type specified is not large enough to hold %d bits.", totalBitLength));
+        }
```

Comment:
This is cool! 
Do we want to support any other types besides INT and LONG? If not, we may want to throw clear exception for that for maximumBits = 0 case.